test/test_other.py

```diff
@@ -11031,6 +11031,24 @@ def check_symbolmap_info(address, func):
     # The name section will not show bar, as it's inlined into main
     check_symbolmap_info(unreachable_addr, '__original_main')
 
+    # 3. Test symbol map on C++ name mangling
+    self.run_process([EMCC, test_file('other/test_symbolmap.cpp'),
+                      '-O1', '--emit-symbol-map', '-o', 'test_symbolmap.js'])
+    self.assertExists('test_symbolmap.js.symbols')
+
+    out_to_js_call_addr = self.get_instr_addr('call\t0', 'test_symbolmap.wasm')
+    unreachable_addr = self.get_instr_addr('unreachable', 'test_symbolmap.wasm')
+
+    def check_cpp_symbolmap_info(address, func):
+      out = self.run_process([emsymbolizer, '--source=symbolmap', '-f', 'test_symbolmap.js.symbols', 'test_symbolmap.wasm', address], stdout=PIPE).stdout
+      self.assertIn(func, out)
+
+    check_cpp_symbolmap_info(out_to_js_call_addr, 'Namespace::foo')         # the function name
+    check_cpp_symbolmap_info(out_to_js_call_addr, 'Namespace::ClassA')      # the parameter
+    check_cpp_symbolmap_info(unreachable_addr, 'Namespace::bar')            # the function name
+    check_cpp_symbolmap_info(unreachable_addr, 'Namespace::ClassA')         # the type parameter
+    check_cpp_symbolmap_info(unreachable_addr, 'Namespace::ClassB')         # the parameter
```

Comment:
Can you also check for `out_to_js`? Names for JS imports is one feature of symbol maps that DWARF and sourcemaps don't have.

Reply:
Sure, but some helps is needed.

How to check an imported JS function, since they don't have wasm code nither do they have address.

We can check if the symbol_map file contains the `out_to_js` string manually, is that OK?

Reply:
Yes, that's what I meant, thanks.